src/main/java/com/microsoft/sqlserver/jdbc/TimeoutCommand.java

```diff
@@ -5,20 +5,28 @@
 
 package com.microsoft.sqlserver.jdbc;
 
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.atomic.AtomicInteger;
+
+
 /**
- * Abstract implementation of a command that can be timed out using the {@link TimeoutPoller}
+ * Abstract implementation of a command that can be timed out using the {@link SQLServerTimeoutManager}
  */
 abstract class TimeoutCommand<T> {
     private final long startTime;
     private final int timeout;
     private final T command;
     private final SQLServerConnection sqlServerConnection;
+    private ScheduledFuture<?> timeoutTask;
+    static AtomicInteger uniqueId = new AtomicInteger();

```

Comment:
im open to suggestions regarding the id stuff, since its theoretically possible for someone to break this limit

Reply:
Let's go for AtomicLong which has a very high range and is safer for applications running for longer time periods.